data/processors/sitemap.py

```diff
@@ -0,0 +1,142 @@
+        # The data's `ranking_factors` have arbitrary scaling, but are for
+        # rooms in general in the range 0 to 9, so we just add 1, divide by 10
+        # and clamp to 1.0 for rooms.
+        # For buildings etc. that are always >= 10, we just subtract 5
+        # to get some kind of relative measure. Also for root (which is
+        # downrated in search) we set 1.0 by hand.
+        if entry["type"] == "root":
+            priority = 1.0
+        elif entry["type"] == "room":
+            priority = min((entry["ranking_factors"]["rank_combined"] + 1) / 10, 1.0)
+        else:
+            priority = min((entry["ranking_factors"]["rank_combined"] - 5) / 10, 1.0)
+
+        sitemaps[sitemap_name].append({
+            "url": url,
+            "last_changed": last_changed,
+            "priority": priority,
+        })
+
+    print(f"{changed_count} of {len(new_data)} URLs have been updated.")
+
+    for name, sitemap in sitemaps.items():
+        _write_sitemap_xml(f"output/sitemap-data-{name}.xml", sitemap)
+
+    _write_sitemapindex_xml(f"output/sitemap.xml", sitemaps)
+
+
+def _download_online_sitemaps(sitemap_names):
+    """ Download online sitemaps by their names """
+    xmlns = "{http://www.sitemaps.org/schemas/sitemap/0.9}"
+    sitemaps = {}
+    for name in sitemap_names:
+        req = urllib.request.Request(f"https://nav.tum.sexy/sitemap-data-{name}.xml")
+        try:
+            with urllib.request.urlopen(req) as resp:
+                sitemap_str = resp.read().decode("utf-8")
+                sitemaps[name] = {}
+                root = defusedET.fromstring(sitemap_str)
+                for child in root.iter(f"{xmlns}url"):
+                    loc = child.find(f"{xmlns}loc")
+                    lastmod = child.find(f"{xmlns}lastmod")
+                    if loc is not None and lastmod is not None:
+                        sitemaps[name][loc.text] = datetime.fromisoformat(lastmod.text)
+        # We gracefully catch any problems (e.g. a 404 HTTPError), because
+        # having the old sitemaps is not critical.
+        except Exception as e:
+            print(f"Failed to download sitemap '{name}': {e}")
+
+    return sitemaps
+
+
+def _write_sitemap_xml(fname, sitemap):
+    """ Write the sitemap XML for a single sitemap """
+    urlset = ET.Element("urlset")
+    urlset.set("xmlns", "http://www.sitemaps.org/schemas/sitemap/0.9")
+    for sitemap_entry in sitemap:
+        url = ET.SubElement(urlset, "url")
+        loc = ET.SubElement(url, "loc")
+        loc.text = sitemap_entry["url"]
+        lastmod = ET.SubElement(url, "lastmod")
+        lastmod.text = sitemap_entry["last_changed"].isoformat()
+        priority = ET.SubElement(url, "priority")
+        priority.text = str(round(sitemap_entry["priority"], 2))
+
+    root = ET.ElementTree(urlset)
+    root.write(fname, encoding="utf-8", xml_declaration=True)
+
+
+def _write_sitemapindex_xml(fname, sitemaps):
```

Comment:
fname is a constant and function is only used once, so there is no real need to pass it. 
It could be named in the docstring.. 
I dont have a strong opinion on this. Current variable passing has upsides as well.

Reply:
It could be constant, though I preferred that the filename strings are all at the same location this way.